plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java

```diff
@@ -32,54 +32,68 @@
 
 package org.opensearch.ingest.attachment;
 
+import org.apache.commons.codec.digest.DigestUtils;
 import org.apache.lucene.tests.util.LuceneTestCase.SuppressFileSystems;
 import org.apache.lucene.tests.util.TestUtil;
 import org.apache.tika.metadata.Metadata;
 import org.opensearch.common.io.PathUtils;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.json.JsonXContent;
 import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
 
 import java.nio.file.DirectoryStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Map;
 
 /**
- * Evil test-coverage cheat, we parse a bunch of docs from tika
- * so that we have a nice grab-bag variety, and assert some content
- * comes back and no exception.
+ * Parse sample tika documents and assert the contents has not changed according to previously recorded checksums.
+ * Uncaught changes to tika parsing could potentially pose bwc issues.
  */
 @SuppressFileSystems("ExtrasFS") // don't try to parse extraN
 public class TikaDocTests extends OpenSearchTestCase {
 
-    /** some test files from tika test suite, zipped up */
+    /** some test files from the apache tika unit test suite with accompanying sha1 checksums */
     static final String TIKA_FILES = "/org/opensearch/ingest/attachment/test/tika-files/";
+    static final String TIKA_CHECKSUMS = "/org/opensearch/ingest/attachment/test/.checksums";
 
-    public void testFiles() throws Exception {
-        Path tmp = createTempDir();
-        logger.debug("unzipping all tika sample files");
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(PathUtils.get(getClass().getResource(TIKA_FILES).toURI()))) {
-            for (Path doc : stream) {
-                String filename = doc.getFileName().toString();
-                TestUtil.unzip(getClass().getResourceAsStream(TIKA_FILES + filename), tmp);
-            }
-        }
-
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(tmp)) {
-            for (Path doc : stream) {
-                logger.debug("parsing: {}", doc);
-                assertParseable(doc);
-            }
-        }
+    @Before
+    public void setLocale() {
```

Comment:
We run tests with randomized locales, please remove default local changes unless there are reasons to do that

Reply:
Tika uses the locale to help inform it's parsing in some cases. In particular testEXCEL.xls and testEXCEL.xlsx will produce different outputs depending on the locale.

Reply:
Our users will be using essentially any locale, we cannot use ENGLISH only, moreover as you said, locale alters the results and we may need to account for that.

Reply:
Would it be appropriate to remove only these two files which have locale dependent parsing? Or alternatively we could validate their output across all possible locales. How would you recommend I handle these locale dependent files while satisfying the initial issue?

Reply:
> Would it be appropriate to remove only these two files which have locale dependent parsing?

That would basically means no checksum check, right? May be we could do this:
 - have a special case of empty checksum, so these files would be parsed but not verified
 - or, make checksum locale dependent, in this case if locale is different from what checksum supports, files would be parsed but not verified

Reply:
Thanks for the input. Removed strict checksum verification of locale dependent sample files.